### ISSUE TITLE
No visible @interface for 'RCTEventEmitter' declares the selector 'invalidate' in RNBackgroundGeolocation.m

### ISSUE THREAD

**Issue by juzer-z** [transistorsoft/react-native-background-geolocation#1407]
While trying to build the iOS app from Xcode, the following error is showing up on line 664 in RNBackgroundGeolocation.m:
No visible @interface for 'RCTEventEmitter' declares the selector 'invalidate'

This leads to inability to build the app or run it. 

## Your Environment
* Plugin version: "^4.4.2"
* Platform: iOS
* OS version: 14.5
* Device manufacturer / model: iPhone 11
* React Native version (`react-native -v`): "^0.63.4"
* Plugin config

## Expected Behavior
Application should have been build and run on the device 

## Actual Behavior
Build Failed with the above error

## Steps to Reproduce

1. Install react-native-background-geolocation as per the guidelines in the doc
2. Run npm i
3. Build application in Xcode

## Context
Simply trying to build and run the app

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
No visible @interface for 'RCTEventEmitter' declares the selector 'invalidate'
```

</details>

**Comment by christocracy** [comment]
Is it a warning or error?

**Comment by christocracy** [comment]
I've got my *SampleApp* on `react-native @ 0.66`.  I seems I made a change [16 days ago to silence a warning.](https://github.com/transistorsoft/react-native-background-geolocation/blame/master/ios/RNBackgroundGeolocation/RNBackgroundGeolocation.m#L664) by calling `[super invalidate]`.

However, [`RCTEventEmitter`](https://github.com/facebook/react-native/blame/main/React/Modules/RCTEventEmitter.m#L89-L98) has had the method `invalidate` for *6 years*.

**Comment by christocracy** [comment]
And when I comment-out that line:
![](https://dl.dropbox.com/s/h7aefd0dt05r4lk/Screen%20Shot%202021-11-10%20at%202.07.38%20PM.png?dl=1)

**Comment by christocracy** [comment]
You're free to comment-out that line if it makes your app build.  I'll continue to investigate.

**Comment by juzer-z** [comment]
It is an error and causing the build to fail.

**Comment by juzer-z** [comment]
> You're free to comment-out that line if it makes your app build. I'll continue to investigate.

Yes, I have done the same as a temporary fix.

**Comment by christocracy** [comment]
Instead of *commenting-out*, this should work.  Let me know and I'll push a new release.

```obj-c
if([super respondsToSelector:@selector(invalidate)]){
    [super invalidate];
}
```

**Comment by juzer-z** [comment]
> ```objc
> if([super respondsToSelector:@selector(invalidate)]){
>     [super invalidate];
> }
> ```

The error persists, 
<img width="756" alt="Screenshot 2021-11-10 at 2 25 01 PM" src="https://user-images.githubusercontent.com/66377168/141179875-d3e804c6-993b-4523-be5e-59bfd24479b6.png">

only commenting the line works.

**Comment by christocracy** [comment]
Clean your build and see if the error disappears

**Comment by juzer-z** [comment]
> Clean your build and see if the error disappears

Yes, cleaned the build and tried to rebuilt. The error persists.

**Comment by christocracy** [comment]
Try this:
```obj-c

if([super respondsToSelector:@selector(invalidate)]){
    [super performSelector:@selector(invalidate)];
}
```

**Comment by christocracy** [comment]
Fix published to `4.4.3`

**Comment by juzer-z** [comment]
> Try this:
> 
> ```objc
> if([super respondsToSelector:@selector(invalidate)]){
>     [super performSelector:@selector(invalidate)];
> }
> ```

Yes this worked

**Comment by ansarikhurshid786** [comment]
I just installed latest version it is not working for me.

**Comment by christocracy** [comment]
> I just installed latest version it is not working for me.

Oh.  That means absolutely nothing.

**Comment by Vikmanatus** [comment]
Hello to everyone :) 

I am running the latest version `4.4.3` on IOS (iPhone 8 14.5.1) I am facing the same error but it is only occuring when I try to run the remote Chrome Debugger

Do you think they could be a quick fix to fix this issue ? :)

**Comment by christocracy** [comment]
@Vikmanatus Thank you for providing *actual information*.  I'm investigating now.

**Comment by Vikmanatus** [comment]
Thank you very much @christocracy, I am deeply sorry I forgot to mention that I am running the project on `react-native` `0.65.1`

**Comment by christocracy** [comment]
Published `4.4.4`

**Comment by Vikmanatus** [comment]
Thank you very much it fixed the issue !

Have a nice day  @christocracy
